plugins/none-ls/_mk-source-plugin.nix

```diff
@@ -0,0 +1,50 @@
+# mkSourcePlugin, returns a module
+sourceType: sourceName:
+{
+  config,
+  pkgs,
+  lib,
+  helpers,
+  ...
+}:
+let
+  inherit (import ./packages.nix pkgs) packaged;
+  pkg = packaged.${sourceName};
+
+  cfg = config.plugins.none-ls;
+  cfg' = config.plugins.none-ls.sources.${sourceType}.${sourceName};
+in
+{
+  options.plugins.none-ls.sources.${sourceType}.${sourceName} =
+    {
+      enable = lib.mkEnableOption "the ${sourceName} ${sourceType} source for none-ls";
+      withArgs = helpers.mkNullOrOption helpers.nixvimTypes.strLua ''
```

Comment:
It should be straighforward (in another PR) to have structured options here no? We'd need some kind of coercion from string to rawLua though

Reply:
Agreed. #1705 is tracking the idea.